proc/basic_server.cpp

```diff
@@ -13,13 +14,39 @@ void error(const char *msg) {
     exit(EXIT_FAILURE);
 }
 
+void *connection_handler(void *socket_desc) {
+    int sockfd = *(int *)socket_desc;
+    char buf[BUFLEN] = {0};
+    char revbuf[BUFLEN] = {0};
+
+    if(read(sockfd, buf, BUFLEN) < 0) {
+        error("ERROR reading from socket\n");
+    }
+
+    // Reverse string
+    int i;
+    int j = 0;
+    int len = strlen(buf);
+    for(i = len - 1; i >= 0; i--) {
+        revbuf[j++] = buf[i];
+    }
+    revbuf[i] = '\0';
+
+    if(write(sockfd, revbuf, strlen(revbuf)) < 0) {
+        error("ERROR writing to socket\n");
+    }
+
+    free(socket_desc);
```

Comment:
see below comment because malloc is causing valgrind to produce warnings.

delete socket_desc;

Reply:
Fixed.